reactor-core/src/test/java/reactor/core/publisher/SinksTest.java

```diff
@@ -0,0 +1,721 @@
+
+			late.assertNoValues().assertErrorMessage("boom");
+		}
+	}
+
+	@Nested
+	class Multicast {
+
+		private Sinks.StandaloneFluxSink<Integer> sink;
+		private Flux<Integer> flux;
+
+		@BeforeEach
+		void createInstance() {
+			sink = Sinks.multicast();
+			flux = sink.asFlux();
+		}
+
+		@Test
+		void fluxViewReturnsSameInstance() {
+			assertThat(flux).isSameAs(sink.asFlux());
+		}
+
+		@Test
+		void isAcceptingMoreThatOneSubscriber() {
+			assertThatCode(() -> {
+				flux.subscribe();
+				flux.subscribe();
+			}).doesNotThrowAnyException();
+		}
+
+		@Test
+		void honorsMultipleSubscribersBackpressure()
+				throws InterruptedException, ExecutionException {
+			ExecutorService es = Executors.newFixedThreadPool(2);
+			CountDownLatch requestLatch = new CountDownLatch(2);
+			final Future<?> f1 = es.submit(() -> {
+				AssertSubscriber<Integer> test1 = AssertSubscriber.create(2);
+				flux.subscribe(test1);
+				test1.assertNoValues();
+				requestLatch.countDown();
+
+				test1.awaitAndAssertNextValues(1, 2);
+				try {
+					Awaitility.await().atMost(2, TimeUnit.SECONDS)
+					          .with().pollDelay(1, TimeUnit.SECONDS)
+					          .untilAsserted(() -> test1.assertValueCount(2));
+				}
+				finally {
+					test1.cancel();
+				}
+			});
+			final Future<?> f2 = es.submit(() -> {
+				AssertSubscriber<Integer> test2 = AssertSubscriber.create(1);
+				flux.subscribe(test2);
+				requestLatch.countDown();
+
+				test2.awaitAndAssertNextValues(1);
+				try {
+					Awaitility.await().atMost(2, TimeUnit.SECONDS)
+					          .with().pollDelay(1, TimeUnit.SECONDS)
+					          .untilAsserted(() -> test2.assertValueCount(1));
+				}
+				finally {
+					test2.cancel();
+				}
+			});
+
+			requestLatch.await(1, TimeUnit.SECONDS);
+			sink.next(1)
+			    .next(2)
+			    .next(3)
+			    .next(4)
+			    .complete();
+
+			f1.get();
+			f2.get();
+		}
+
+		@Test
+		void doesNotReplayToLateSubscribers() {
+			AssertSubscriber<Integer> s1 = AssertSubscriber.create();
+			AssertSubscriber<Integer> s2 = AssertSubscriber.create();
+
+			flux.subscribe(s1);
+			sink.next(1).next(2).next(3);
+			s1.assertValues(1, 2, 3);
+
+			flux.subscribe(s2);
+			s2.assertNoValues().assertNotComplete();
+
+			sink.complete();
+			s1.assertValueCount(3).assertComplete();
+			s2.assertNoValues().assertComplete();
+		}
+
+		@Test
+		public void doesBufferBeforeFirstSubscriber() {
+			AssertSubscriber<Integer> first = AssertSubscriber.create();
+
+			sink.next(1).next(2).next(3);
+			flux.subscribe(first);
+
+			first.assertValues(1, 2, 3).assertNotComplete();
+
+			sink.complete();
+			first.assertComplete();
+		}
+
+		@Test
+		public void replayAndCompleteFirstSubscriber() {
+			AssertSubscriber<Integer> first = AssertSubscriber.create();
+
+			sink.next(1).complete();
+			flux.subscribe(first);
+
+			first.assertValues(1).assertComplete();
+		}
+
+		@Test //TODO is that acceptable that EmitterProcessor doesn't replay in case of errors?
```

Comment:
Probably not. Especially if `onErrorResume` is used on the returning `Flux`

Reply:
isolated that in the test a bit more, will have to decide what to do by GA